### ISSUE TITLE
Add ability to duplicate whole event page to quickly populate repetitive adding of events

### ISSUE THREAD

**Issue by willguv** [localgovdrupal/localgov_events#96]
This was posted by @LizziCBFC 

> We run a lot of events that are very similar. When we get blocks of events to add we are manually creating an event for every one, and the location/facet details are often the same. It would be useful to be able to quickly duplicate a single event and tweak where required.

**Comment by andybroomfield** [comment]
Maybe something like
- [Quick Node Clone](https://www.drupal.org/project/quick_node_clone)
- [Entity Clone](https://www.drupal.org/project/entity_clone)
- [Clone Nodes](https://www.drupal.org/project/clone_nodes)

How will this play with the recurring date functionlity that already exists with events?
I seem to remember cloning nodes causes some issues with refrences as just the references used to get cloned, but still point to underlying data.

**Comment by ekes** [comment]
We're using [Replicate](https://drupal.org/project/replicate) for microsites. It's much easier to configure (as a programmer) and hide the complexity from the user. Some of the other clone options have quite scary UIs to default clone paragraphs, but not taxonomy terms etc. _But_ using replicate would need thought here if other fields can be added to the content type by the end implementation.

**Comment by andybroomfield** [comment]
Sounds like a good plan, though is replicate API only?

**Comment by ekes** [comment]
> Sounds like a good plan, though is replicate API only?

Yes, exactly the reason you can end up with a decent UI :) What I've done elsewhere with it (for user triggered clone of individual content) is add a route, local task and controller (extending drupal confirm form). This clones the entity and appropriate referenced entities (down the tree as required).

It's just the thing about adding fields on different sites. You will end up needing a configuration UI that's like Entity Clone? Or is there a more clever way of doing that too, when we know a lot about the entities already?

**Comment by andybroomfield** [comment]
For reference #56 has the differences between stock localgov_events and what Brighton has, so some way to either extend it or for it to know to clone all fields and how to treat them would be helpful.
I guess from a UI perspective its just replicate event and takes you to the new events edit page when done.

**Comment by ekes** [comment]
> I guess from a UI perspective its just replicate event and takes you to the new events edit page when done.

Exactly, removing the option to choose which bits get cloned makes it with the example I mentioned: click 'Clone' tab, hit 'Confirm' on form, end up on new cloned content.

**Comment by willguv** [comment]
Thanks for this input everyone!

"Duplicate" seems a better word than "Clone", do we have any choice over this?

Flagging to @LizziCBFC too

**Comment by willguv** [comment]
@RyanFH257 uses Entity Clone at H&F. Could bulk upload of events remove the need for cloning

@andybroomfield familiar for events in Libraries etc. If we use Replicate we could configure something that makes sense for each content type

@stephen-cox agrees Replicate best for this reason

Will: we need to avoid people having to think about what part of the content they need to clone. We can create sensible defaults

eg clone an event, but not the venue

@stephen-cox let's estimate effort Entity Clone vs Replicate

**Comment by millnut** [comment]
Just adding on to this, but Drupal Starshot/CMS they use [Quick Node Clone](https://www.drupal.org/project/quick_node_clone)
